lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -1007,7 +1007,11 @@ DIFile *SPIRVToLLVMDbgTran::getFile(const SPIRVId SourceId) {
          "DebugSource instruction is expected");
   SPIRVWordVec SourceArgs = Source->getArguments();
   assert(SourceArgs.size() == OperandCount && "Invalid number of operands");
-  StringRef Checksum(getString(SourceArgs[TextIdx]));
+  std::string ChecksumStr =
+      !getDbgInst<SPIRVDebug::DebugInfoNone>(SourceArgs[TextIdx])
+          ? getString(SourceArgs[TextIdx])
+          : "";
+  StringRef Checksum(ChecksumStr);
```

Comment:
I think we can skip allocation of extra `StringRef` variable. I guess `StringRef` can be constructed from `std::string` implicitly if we pass `ChecksumStr` to `ParseChecksum()`